File: trees/typescript/recorridos/MainTree3.ts
import { Tree3 } from "./Tree3";


const objTree3 = new Tree3(
  "a",
  new Tree3(
    "b",
    new Tree3(
      "d",
      new Tree3("j", null, null, null),
      new Tree3("k", null, null, null),
      null
    ),
    new Tree3("e", null, null, null),
    null
  ),
  new Tree3(
    "c",
    new Tree3("f", null, null, null),
    new Tree3("g", null, null, null),
    null
  ),
  new Tree3(
    "b",
    new Tree3("h", null, null, null),
    new Tree3(
      "i",
      new Tree3("m", null, null, null),
      null,
      null
    ),
    null
  )
)

console.log("Pre-Order: ", objTree3.preorder())
console.log("In-Order: ", objTree3.inorder())
console.log("Pos-Order: ", objTree3.posorder())

File: trees/python/recorridos/MainTree3.py
from Tree3 import Tree3


objTree3 = Tree3(
    "a",
    Tree3(
        "b",
        Tree3("e", None, None, None),
        Tree3("f", None, None, None),
        Tree3("g", None, None, None)
    ),
    Tree3(
        "c",
        Tree3("h", None, None, None),
        None,
        None
    ),
    Tree3("d", None, None, None)
)

objTree3b = Tree3(
    "a",
    Tree3(
        "b",
        Tree3(
            "d",
            Tree3("j", None, None, None),
            Tree3("k", None, None, None),
            None
        ),
        Tree3("e", None, None, None),
        None
    ),
    Tree3(
        "c",
        Tree3("f", None, None, None),
        Tree3("g", None, None, None),
        None
    ),
    Tree3(
        "b", 
        Tree3("h", None, None, None),
        Tree3(
            "i",
            Tree3("m", None, None, None),
            None,
            None
        ),
        None
    )
)

print(objTree3.preorder())
print(objTree3.inorder())
print(objTree3.posorder())

print("\nOtro Arbol")

print(objTree3b.preorder())
print(objTree3b.inorder())
print(objTree3b.posorder())
